### PR TITLE
Cleanup docker driver

### PR DESCRIPTION
**Pull request by cbednarski**
- Improved a lot of logging statements to have more information and better phrasing
- DRY up some code in network configuration
- Add config.ReadBool and config.ReadBoolDefault

### CODE REVIEW

File: client/driver/docker.go [hunk 1]
```diff
@@ -213,24 +217,15 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task, dri
...
+		return c, fmt.Errorf(`Unable to set privileged flag since "docker.privileged.enabled" is false`)
 	}
+	hostConfig.Privileged = hostPrivileged
 
 	// set DNS servers
-	dns := driverConfig.DNS
-
-	if dns != "" {
-		for _, v := range strings.Split(dns, ",") {
+	if driverConfig.DNS != "" {
```
Comment (dadgar) [inline client/driver/docker.go]: can the DNS field just be changed to []string

File: client/driver/docker.go [hunk 2]
```diff
@@ -241,10 +236,8 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task, dri
 	}
 
 	// set DNS search domains
-	dnsSearch := driverConfig.SearchDomains
-
-	if dnsSearch != "" {
-		for _, v := range strings.Split(dnsSearch, ",") {
+	if driverConfig.SearchDomains != "" {
```
Comment (dadgar) [inline client/driver/docker.go]: Same

File: client/driver/docker.go [hunk 3]
```diff
@@ -268,7 +261,10 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task, dri
 
 	// Setup port mapping and exposed ports
 	if len(task.Resources.Networks) == 0 {
-		d.logger.Print("[WARN] driver.docker: No network resources are available for port mapping")
+		d.logger.Print("[INFO] driver.docker: No network interfaces are available")
```
Comment (dadgar) [inline client/driver/docker.go]: This debug message is only interesting if they want port forwarding. If I don't ask for any ports then I don't really care about this message. Also if you do decide to keep it, Print -> Println

File: client/driver/docker.go [hunk 4]
```diff
@@ -276,11 +272,14 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task, dri
 		exposedPorts := map[docker.Port]struct{}{}
 
 		for _, port := range network.ReservedPorts {
-			publishedPorts[docker.Port(strconv.Itoa(port.Value)+"/tcp")] = []docker.PortBinding{docker.PortBinding{HostIP: network.IP, HostPort: strconv.Itoa(port.Value)}}
-			publishedPorts[docker.Port(strconv.Itoa(port.Value)+"/udp")] = []docker.PortBinding{docker.PortBinding{HostIP: network.IP, HostPort: strconv.Itoa(port.Value)}}
+			portStr := docker.Port(port.Value)
+
+			publishedPorts[portStr+"/tcp"] = []docker.PortBinding{docker.PortBinding{HostIP: network.IP, HostPort: strconv.Itoa(port.Value)}}
```
Comment (dadgar) [inline client/driver/docker.go]: Do we need the strconv.Itoa?

Comment (cbednarski) [inline client/driver/docker.go]: Hm. `portStr` is actually a docker port type wrapper for string type, so this might work.

File: client/driver/docker.go (hunk 1 above)
Comment (cbednarski) [inline client/driver/docker.go]: Yes. I plan to make that change in a separate PR since it changes the config syntax.

File: client/driver/docker.go (hunk 4 above)
Comment (cbednarski) [inline client/driver/docker.go]: Compiler says no:
```
client/driver/docker.go:277: cannot use portStr (type docker.Port) as type string in field value
client/driver/docker.go:278: cannot use portStr (type docker.Port) as type string in field value
```
